src/test/java/stormpot/CompoundExpirationTest.java

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package stormpot;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class CompoundExpirationTest {
+
+    private static final Expiration<Poolable> alwaysExpired = new StaticExpiration(true);
+    private static final Expiration<Poolable> neverExpired = new StaticExpiration(false);
+
+    @Test public void
+    expiresWhenBothExpirationsExpire() throws Exception {
+        CompoundExpiration<Poolable> compoundExpiration = new CompoundExpiration<Poolable>(alwaysExpired, alwaysExpired);
+
+        assertTrue(compoundExpiration.hasExpired(dummySlotInfo()));
+    }
+
+    @Test public void
+    expiresWhenOneExpirationExpires() throws Exception {
+        CompoundExpiration<Poolable> compoundExpiration = new CompoundExpiration<Poolable>(alwaysExpired, neverExpired);
+
+        assertTrue(compoundExpiration.hasExpired(dummySlotInfo()));
+
+        compoundExpiration = new CompoundExpiration<Poolable>(neverExpired, alwaysExpired);
+
+        assertTrue(compoundExpiration.hasExpired(dummySlotInfo()));
+    }
+
+    @Test public void
+    doesNotExpireWhenNoExpirationExpire() throws Exception {
+        CompoundExpiration<Poolable> compoundExpiration = new CompoundExpiration<Poolable>(neverExpired, neverExpired);
+
+        assertFalse(compoundExpiration.hasExpired(dummySlotInfo()));
+    }
+
+    private SlotInfo<Poolable> dummySlotInfo() {
+        return new DummySlotInfo();
+    }
+
+    private static final class StaticExpiration implements Expiration<Poolable> {
+
+        private final boolean expiration;
+
+        private StaticExpiration(boolean expiration) {
+            this.expiration = expiration;
+        }
+
+        @Override
+        public boolean hasExpired(SlotInfo<? extends Poolable> info) throws Exception {
+            return expiration;
+        }
+    }
+
+    private static final class DummySlotInfo implements SlotInfo<Poolable> {
+        @Override
+        public long getAgeMillis() {
+            return 0;
+        }
+
+        @Override
+        public long getClaimCount() {
+            return 0;
+        }
+
+        @Override
+        public Poolable getPoolable() {
+            return null;
+        }
+
+        @Override
+        public int randomInt() {
+            return 0;
+        }
+
+        @Override
+        public long getStamp() {
+            return 0;
+        }
+
+        @Override
+        public void setStamp(long stamp) {}
+    }
```

Comment:
The `TimeSpreadExpirationTest` and `TimeExpirationTest` have similar classes. They could be pulled out and consolidated into one.